### PR TITLE
LLVM bump (as we know it)

### PR DESCRIPTION
**Pull request by drom**
It was hard.
80% of credit goes to @youngar

- `func` and `call` op parsing requires the dialect specifier: https://discourse.llvm.org/t/psa-special-case-parsing-of-func-operations-is-being-removed/61953
- Region argument parsing overhaul: https://reviews.llvm.org/D124649
- LLVM requires a cmake build type to be specified: https://reviews.llvm.org/D124153

### CODE REVIEW

File: lib/Dialect/Calyx/CalyxOps.cpp [hunk 1]
```diff
@@ -525,7 +526,9 @@ ParseResult ComponentOp::parse(OpAsmParser &parser, OperationState &result) {
                              result.attributes))
     return failure();
 
-  SmallVector<OpAsmParser::UnresolvedOperand> ports;
+  // SmallVector<OpAsmParser::UnresolvedOperand> ports;
```
Comment (youngar) [inline lib/Dialect/Calyx/CalyxOps.cpp:529]:
```suggestion
```

File: lib/Dialect/FIRRTL/FIRRTLOps.cpp [hunk 2]
```diff
@@ -894,6 +896,10 @@ parseModulePorts(OpAsmParser &parser, bool hasSSAIdentifiers,
       return failure();
     portTypes.push_back(TypeAttr::get(portType));
 
+    if (hasSSAIdentifiers) {
+      entryArgs.back().type = portType;
+    }
```
Comment (youngar) [inline lib/Dialect/FIRRTL/FIRRTLOps.cpp:901]:
```suggestion
    if (hasSSAIdentifiers)
      entryArgs.back().type = portType;
```

File: lib/Dialect/MSFT/MSFTOps.cpp [hunk 3]
```diff
@@ -425,12 +423,15 @@ ParseResult MSFTModuleOp::parse(OpAsmParser &parser, OperationState &result) {
...
-          resultAttrs, resultNames))
+          parser, entryArgs, isVariadic, resultTypes, resultAttrs, resultNames))
     return failure();
 
   // Record the argument and result types as an attribute.  This is necessary
   // for external modules.
+  SmallVector<Type> argTypes;
+  for (auto arg : entryArgs)
+    argTypes.push_back(arg.type);
+
```
Comment (youngar) [inline lib/Dialect/MSFT/MSFTOps.cpp:434]:
```suggestion
```

**Review by teqdruid — APPROVED** [review]
The ESI and MSFT changes look good to me. So do the workflow changes. nit: the white space fixes are nice, but unrelated to the llvm bump.

**Review by darthscsi — APPROVED** [review]

**Pull request merged**